Update to Qt port

JensReimann opened the pull request:
overall improvements and bug fixes to Qt port

JensReimann committed:
- update Qt port to latest VCL version

Commit:
Merge remote-tracking branch 'RTKLIB/rtklib_2.4.3' into rtklib_2.4.3

merge takasu's rtklib_2.4.3 b16

JensReimann committed:
update Qt port to takasu's rtklib_2.4.3 b16

- add include <clocale>
- some update project files

JensReimann committed:
suppress lots of warning

JensReimann committed:
suppress more warnings
remove unused function from septentrio.c

JensReimann committed:
Big Patch

- remove Qt4 support
- change may input from plain linedits to integer or double spin boxes
- use joint date & time input
- many cosmetic changes

JensReimann committed:
- small cleanup

JensReimann committed:
- add Visual C profile to project

JensReimann committed:
- improve build system

JensReimann committed:
- fix typo

JensReimann committed:
- remove no longer needed code

JensReimann committed:
bug fixes

- zoom in rtkplotqt
- date/time input widget time zone
- rtkpost_qt "abort" action

JensReimann committed:
update to latest version (2.4.3 R19)

JensReimann committed:
Updates

including:
- Improve scaling of window content
- fix milli seconds in QDateTime to gtime_t conersion
- fixes in srctblbrowser_qt
- add missing icons to resource files

JensReimann committed:
Merge remote-tracking branch 'Main/rtklib_2.4.3' into rtklib_2.4.3

JensReimann committed:
Update Qt code to latest changes in mainstream

JensReimann committed:
Merge remote-tracking branch 'Main/rtklib_2.4.3' into rtklib_2.4.3

JensReimann committed:
Update to latest version

JensReimann committed:
Merge remote-tracking branch 'Main/rtklib_2.4.3' into rtklib_2.4.3

JensReimann committed:
update to rtklib_2.4.3 b25

JensReimann committed:
Merge remote-tracking branch 'Main/rtklib_2.4.3' into rtklib_2.4.3

JensReimann committed:
update to rtklib_2.4.3 b26

JensReimann committed:
- small fix in strsvr_qt

JensReimann committed:
- Septentrio: delete removed signal frequencies and use rtklib constant
for glonass L3
- small bug fixes

JensReimann committed:
- fixed "NTRIP caster User-ID can't contain" (#5)
- update tab order in all dialogs
- fix mssing ico file on windows

JensReimann committed:
- fix stop button in strsvr_qt

JensReimann committed:
- fix display of RTCM messages in monitor dialog
- fix scroll down button in monitor dialog
- fix possible crash on shuwdown if monitor dialog is open

JensReimann committed:
add raw output to septentrio driver

JensReimann committed:
- fix scaling buttons in rtknavi_qt
- fix jumps at start of mouse dragging action

JensReimann committed:
fix periodic command sending causing issue #8

JensReimann committed:
- beautifying

JensReimann committed:
- fixed handling of additional start and stop commands to be send
- fixed so memory holes

JensReimann committed:
- fix memory holes

JensReimann committed:
- fix native suport of pathes using drag'n'drop
- fix alignment of widget in rtknavi_qt option dialog

JensReimann committed:
Fix to tcp option dialog

JensReimann committed:
- fix drag'n'drop issue in all remaining apps

JensReimann committed:
- UI improvement
- fix reading of minimum elevation angle from config file

JensReimann committed:
- fix GUI issu
	- checkbox was not updating control elements in cmdoptdlg
	- GLONASS issue and IODE/IODC were not displayed in monitor dialog of rtknavi_qt)
- minor cleanup in septentrio support (IODC is not transmitted at all
for Galileo and Galileo code is now correctly stored)

JensReimann committed:
Merge remote-tracking branch 'Main/rtklib_2.4.3' into rtklib_2.4.3

JensReimann committed:
update to b27

JensReimann committed:
- fix typos
- add some new .gitignore file
- add missing project files

JensReimann committed:
Merge remote-tracking branch 'remotes/Main/rtklib_2.4.3' into rtklib_2.4.3

JensReimann committed:
- bump to b28

JensReimann committed:
- improve libiers compilation

JensReimann committed:
- small cleanup in septentrio receiver support

JensReimann committed:
-update .gitignore

JensReimann committed:
fix station file selection in rtkpost_qt option dialog

JensReimann committed:
- fix types in rtkconv_qt which causes input files 8+9 not working

JensReimann committed:
Update to b29

Merge remote-tracking branch 'Main/rtklib_2.4.3' into rtklib_2.4.3

JensReimann committed:
-make use of IERS model an option (requires a fortran compiler)
-small fix in septrentrio driver

JensReimann committed:
- force c++11 needed for clang compiler

JensReimann committed:
Merge branch 'rtklib_2.4.3' of https://github.com/JensReimann/RTKLIB into rtklib_2.4.3

JensReimann committed:
fix startup tray option in rtknavi_qt

JensReimann committed:
Merge remote-tracking branch 'remotes/Main/rtklib_2.4.3' into rtklib_2.4.3

JensReimann committed:
adapted Qt version to patch level p31

JensReimann committed:
- fix save dialog in rtknavi-qt option dialog (was calling an open dialog instead)

JensReimann committed:
- limit fresh rate for graph in rtkplot_qt
- minor optimization in drawing code
- minor fixes

JensReimann committed:
- fix typo in rtknavi_qt/navimain.cpp causing rtkplot_qt call to fail

JensReimann committed:
- fix path to google maps file

JensReimann committed:
- don't write a new file when inserting API key for google maps view
- improve view alignment for GM and GE

JensReimann committed:
- load html code for google maps from header file instead of using a dedicated html file

JensReimann committed:
Merge remote-tracking branch 'Main/rtklib_2.4.3' into rtklib_2.4.3

JensReimann committed:
- bump to version b32

holishing commented:
I'm curious about if it is possible to cross-compile win32/win64 binary in Linux (for e.g., use MXE or dockcross) without another major modification of this Qt port? Thanks!

holishing commented:
alright, I've use MXE (mingw version qt5 and mingw version gcc 9.1.0) build out rtkget_qt, thanks for your work on Qt porting!

holishing commented:
win32 executable can be built in Qt5 (in little patch) in 2.4.3 branch in [this repository](https://github.com/tomojitakasu/RTKLIB):
- [x] rtkget_qt.exe
- [x] rtkpost_qt.exe
- [x] rtklaunch_qt.exe

~others need to be figure out problems in compile progress.~
other programs can only pass cross-compilation in [Qt port repository](https://github.com/JensReimann/RTKLIB) in [little patch](https://github.com/JensReimann/RTKLIB/pull/21).  I've sent PR to that repo.

Here's my own Win32 binary build for Qt Ports, if you trust this method, it is welcome to test!
https://github.com/holishing/RTKLIB/releases/tag/2.4.3-b32-p1

holishing committed:
fix ws2_32 library name for case-sensitive

change Ws2_32 to ws2_32 for case-sensitive problem that makes link error in cross-compile environment for MinGW (Win32 excutable) by MXE toolchain in linux platform

MXE: https://mxe.cc/

JensReimann committed:
Merge pull request #21 from holishing/rtklib_2.4.3_b32p1

fix ws2_32 library name for case-sensitive

JensReimann committed:
- fix font selection dialog in rtkplot_qt

JensReimann committed:
Merge branch 'rtklib_2.4.3' of https://github.com/JensReimann/RTKLIB into rtklib_2.4.3

JensReimann committed:
- undo font selection fix for macOS

lbartoletti commented:
What is the status of this PR?

ValZapod commented:
Yeah, what is the status?

JensReimann commented:
No pull yet. The code lives in a separate repository for now:
https://github.com/JensReimann/RTKLIB/tree/rtklib_2.4.3

JensReimann committed:
Merge branch 'rtklib_2.4.3' of https://github.com/tomojitakasu/RTKLIB into rtklib_2.4.3

tomojitakasu commented:
Thanks for excellent work to port RTKLIB Windows APs to Qt.
The directory RTKLIB/app/*_qt are just moved to RTKLIB/app/qtapp in 2.4.3 b34 without modification.
I plan to remove Qt APs in the trunk repository in the next beta release or formal 2.4.3 release in spring because:
(1) The size of the trunk repository should be small and compact as long as possible.
(2) Someone reported about quality issues in Qt APs but it seems hard to maintain them simultaneously as same as other primary codes.
So please consider to move Qt APs to a forked repository outside of the trunk repository.

JensReimann commented:
Hello Tomoji-san,
I support your decision to remove the Qt frontend from your repository. It 
could never keep track with the changes you implemented for the rest of 
rtklib. 
Thee is already a dedicated repository (https://github.com/JensReimann/
RTKLIB/) which is adapted to recent version of rtklib (currently up to 2.4.3 
beta 33).
Feel free to remove the outdated code.

Best regards,
Jens
 
Am Mittwoch, 6. Januar 2021, 04:18:41 CET schrieb Tomoji Takasu:
> Thanks for excellent work to port RTKLIB Windows APs to Qt.
> The directory RTKLIB/app/*_qt are just moved to RTKLIB/app/qtapp in 2.4.3
> b34 without modification. I plan to remove Qt APs in the trunk repository
> in the next beta release or formal 2.4.3 release in spring because: (1) The
> size of the trunk repository should be small and compact as long as
> possible. (2) Someone reported about quality issues in Qt APs but it seems
> hard to maintain them simultaneously as same as other primary codes. So
> please consider to move Qt APs to a forked repository outside of the trunk
> repository.

tomojitakasu commented:
Dear Jens,

I'm really grateful for your contribution. I understand the porting is not easy work. It could not be done without your great effort.

Thanks again for understanding.

The link to the forked repository will be kept in the next release.

regards,

ValZapod commented:
QT should be supported in mainline, IMHO.
 
@JensReimann The code is here https://github.com/tomojitakasu/RTKLIB/tree/rtklib_2.4.3/app/qtapp

thesamprice commented:
I also vote for qt so it can run on mac / Linux

On Thu, Jan 7, 2021 at 6:40 PM Valerii Zapodovnikov <
notifications@github.com> wrote:

> QT should be supported in mainline, IMHO.
>
> @JensReimann <https://github.com/JensReimann> The code is here
> https://github.com/tomojitakasu/RTKLIB/tree/rtklib_2.4.3/app/qtapp
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tomojitakasu/RTKLIB/pull/261#issuecomment-756453226>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAGHUMA5CVTAFIESJQCNGELSYZBANANCNFSM4C2XDTTA>
> .
>
-- 
Thank you,

Sam Price
(707) 742-3726

DavidKelleySCSC commented:
> I also vote for qt so it can run on ma

Me too.   **Big up-vote** on that topic. 

My needs are a bit different, but I would like to see further revisions all move away form the current IDE that is a dead end to some.  We have been using Qt for roughly ten years now along with RTKLIB parts (having gotten there from the BKG folks convincing us it was a good choice back in the day). Now we have the keep two copies in some of the places where we use RTKLIB due to that.  Jens has done great thing to move the code in that direction, how can we move QT to the mainline in this?

JensReimann committed:
Merge remote-tracking branch 'remotes/rtklib/rtklib_2.4.3' into rtklib_2.4.3

JensReimann committed:
update to b34

JensReimann committed:
finish merge

Commit:
Add file extensions for RINEX v3 filenames

Commit:
Remove duplicate inclusion of widget

Commit:
Ignore directories for binaries

Commit:
Use bin directory in user home

Commit:
Add master makefile

Commit:
Ignore .qmake.stash

Commit:
Fix typos

Commit:
Fix typos

Commit:
Fix typos

Commit:
Activated compilation of tools in util

Commit:
Add clean-up of old Qt makefiles

Commit:
Fix typos

Commit:
Fix signal list in RINEX header

Commit:
Use bin folder in RTKLIB root directory

Commit:
Fix typos

Commit:
Fix typos

Commit:
Add navigation system selection to command line options, remove obsolete
variables

Commit:
Fix typos

Commit:
Fix typos

Commit:
Fix typo

Commit:
Fix typos

Commit:
Fix typos

Commit:
Major revision

Commit:
Fix typos

Commit:
Fix typo

Commit:
Revised satellite position computation

Commit:
Fix typos

Commit:
Add signal selection through command line arguments

Commit:
Remove -sys command line argument

Commit:
Add executables in util folder to gitignore

Commit:
Change logging output

Commit:
Fix GLONASS frequency index extraction in outrnx_glo_fcn()

Commit:
Fix typo

Commit:
Improve selection of observation types

Commit:
Fix typos

Commit:
Add measurement noise

Commit:
Add simulation options data structure

Commit:
Fix typo

JensReimann committed:
fix a bug (too short buffer) in rtkplot_qt causing wrong time timestamps
and consequently wrong plotting

AndreHauschild committed:
Merge pull request #1 from JensReimann/rtklib_2.4.3

fix a bug (too short buffer) in rtkplot_qt causing wrong time timestamps

Commit:
Fix command line option for ini file

Commit:
Remove obsolete whitespace

Commit:
Remove duplicate call to LoadOpt()

Commit:
Add inclusion of tides.c

Commit:
Fix typo

Commit:
Add earth tides correction

JensReimann committed:
- fix another potential invalid access (reported and proposed fix provided by Florian Wobbe)

Commit:
Fix typos

Commit:
Fix IERS reference

AndreHauschild committed:
Merge branch 'JensReimann:rtklib_2.4.3' into rtklib_2.4.3

Commit:
Enable processing without RINEX-NAV, fix receiver position without ERP

Commit:
Add command line options for noise, iono and tropo

Commit:
Return values also for zero clock offset

Commit:
Add seperate --tides option

Commit:
Fix typos

Commit:
Add help text and other command line options

Commit:
Add compilation of unit-tests

Commit:
Enable Galileo

Commit:
Add unit test executables

Commit:
Update antex file

Commit:
Use lib.pro

Commit:
Restored compilation of iers lib

Commit:
Change console app install directory

Commit:
Add debugging output

Commit:
Merge branch 'rtklib_2.4.3' of github.com:AndreHauschild/RTKLIB into rtklib_2.4.3

Commit:
Fix test data folder path

Commit:
Add PPP test case

Commit:
Fix bug for iono delay computation in udbias_ppp()

Commit:
Add eop and blq file options

Commit:
Fix typos

Commit:
Fixed code alignment

Commit:
Enable output of ambiguities

Commit:
Enable compilation with -DIERS_MODEL

Commit:
Fix typo

Commit:
Fix code alignment

Commit:
Enable compilation with -DIERS_MODEL

JensReimann committed:
- fix plotting issue pointed out by alexngi66

AndreHauschild committed:
Merge pull request #2 from JensReimann/rtklib_2.4.3

- fix plotting issue pointed out by alexngi66

Commit:
Add check for required input for iono modeling

Commit:
Fix typos

AndreHauschild committed:
Revert "Change console app install directory"

This reverts commit 046f926a0ee661a2372c7e1dc069f25edea9f84e.

AndreHauschild committed:
Revert "Return values also for zero clock offset"

This reverts commit 66841de21e8b2f29ec4b34f66735e65a5aeaf1f9.

AndreHauschild committed:
Revert "Use bin directory in user home"

This reverts commit 784dd85f6520476730a2566b669838d7aeb12e76.

JensReimann committed:
Merge pull request #30 from AndreHauschild/rtklib_2.4.3

Accumlated commits with updates and typo corrections

JensReimann committed:
adapt formally to Qt6

JensReimann committed:
- cleanup in Qt gui code
